include/cosim/timer.hpp

```diff
@@ -36,6 +36,15 @@ struct real_time_config
      * This value is used for monitoring purposes only.
      */
     std::atomic<int> steps_to_monitor = 5;
+
+    /**
+     *  A sampling period in second(s) used in the rolling average real time factor calculation.
+     *  This can be useful when simulation step size is small and a fixed value for `steps_to_monitor`
+     *  would not compute an accurate rolling average real time factor.
+     *  When the value is greater than zero, the real time factor is computed periodically using the
+     *  specified time instead of the `steps_to_monitor` value.
+     */
+    std::atomic<double> sampling_period_to_monitor = -1;
```

Comment:
Have you considered using a `std::chrono::duration` type for this, for example `std::chrono::milliseconds` (or `std::chrono::duration<double>` if it really needs to be floating-point)? I think it would express the semantics better, though I don't know whether it will work with `std::atomic`.

Reply:
I also  considered using `std::chrono::duration` but the `std::atmoic` documentation says:

> The primary std::atomic template may be instantiated with any [TriviallyCopyable](https://en.cppreference.com/w/cpp/named_req/TriviallyCopyable) type T satisfying both [CopyConstructible](https://en.cppreference.com/w/cpp/named_req/CopyConstructible) and [CopyAssignable](https://en.cppreference.com/w/cpp/named_req/CopyAssignable). The program is ill-formed if any of following values is false
    - [std::is_trivially_copyable](http://en.cppreference.com/w/cpp/types/is_trivially_copyable)<T>::value
    - [std::is_copy_constructible](http://en.cppreference.com/w/cpp/types/is_copy_constructible)<T>::value
    - [std::is_move_constructible](http://en.cppreference.com/w/cpp/types/is_move_constructible)<T>::value
    - [std::is_copy_assignable](http://en.cppreference.com/w/cpp/types/is_copy_assignable)<T>::value
    - [std::is_move_assignable](http://en.cppreference.com/w/cpp/types/is_move_assignable)<T>::value

Wasn't sure if this is the case for `std::chrono::duration` so just used `double` indicating seconds.

We can use mutex and create synchronized getter/setters after making the var private. Not sure how this would affect on the performance over `std::atomic` (since mutex locking/unlocking would be called on every step).

Reply:
I'm pretty sure `std::chrono::duration` is trivially copyable, so it should work.